scipy/_lib/_array_api.py

```diff
@@ -339,3 +339,7 @@ def cov(x, *, xp=None):
 
 def xp_unsupported_param_msg(param):
     return f'Providing {param!r} is only supported for numpy arrays.'
+
+
+def is_complex(x, xp):
+    return x.dtype in [xp.complex64, xp.complex128]
```

Comment:
We could do `xp.isdtype(x.dtype, 'complex floating')`, but this seems robust enough to me. Unless `complex256` is likely to appear in the standard any time soon??

Reply:
Since NumPy has `complex256` and PyTorch has `complex32` I think we should be more precise here. Either use `xp.isdtype` or assert the dtype is one of the ones in the array api standard.

Reply:
> 'complex floating': complex floating-point data types (e.g., complex64, complex128).

Since the standard uses 'e.g.' here, this should be more robust and cover all complex dtypes. So I'll go with `xp.isdtype`.

Reply:
The standard is unlike to gain more complex dtypes, but use of `isdtype` seems like a good idea either way.

Reply:
Addressed https://github.com/scipy/scipy/pull/19263/commits/24774d0e999ba49146ca8434586ea75b82227b37.